src/RMGGeneratorDecay0.cc

```diff
@@ -43,4 +50,129 @@ void RMGGeneratorDecay0::GeneratePrimaries(G4Event* event) {
+    // Sanity check
+    if (seed <= 0) {
+      RMGLog::Out(
+          RMGLog::error,
+          "CLHEP seed seems invalid or uninitialized. Setting BxDecay0 seed to 0."
+      );
+      seed = 0;
+    }
+
+    // This should never occur on most systems, as the only way is through the internal CLHEP table
+    // and the values there are all below the max int32.
+    if (seed > std::numeric_limits<int>::max()) {
+      int new_seed = seed - std::numeric_limits<int>::max();
+      RMGLog::Out(
+          RMGLog::warning,
+          "Seed ",
+          seed,
+          " is too large for BxDecay0. Largest possible seed is ",
+          std::numeric_limits<int>::max(),
+          ". Setting BxDecay0 seed to ",
+          new_seed
+      );
+      seed = new_seed;
+    }
+
+    bxdecay0_g4::PrimaryGeneratorAction::ConfigurationInterface&
+        configInt = fDecay0G4Generator->GrabConfiguration();
+    configInt.seed = seed;
+    RMGLog::Out(RMGLog::debug, "BxDecay0 generator: seed set to ", seed);
+    fDecay0G4Generator->SetConfigHasChanged(true);
+  }
+}
+
+void RMGGeneratorDecay0::SetBackground(std::string isotope) {
+  RMGLog::Out(RMGLog::debug, "Setting BxDecay0 background mode with isotope: ", isotope);
+  bxdecay0_g4::PrimaryGeneratorAction::ConfigurationInterface& configInt = fDecay0G4Generator
+                                                                               ->GrabConfiguration();
+  configInt.reset_base();
+  configInt.decay_category = "background";
+  configInt.nuclide = isotope;
+  configInt.debug = false; // If we want a debug mode, we need to add a command for it
+  SetUpdateSeeds(true);    // The seed and config update should now be done in BeginOfRunAction
+}
+
+void RMGGeneratorDecay0::DefineCommands() {
+
+  fMessenger = std::make_unique<G4GenericMessenger>(
+      this,
+      "/RMG/Generator/BxDecay0/",
+      "Commands for controlling the BxDecay0 generator"
+  );
+
+  // Need an additional command because both modes expect different parameters.
+  fMessenger->DeclareMethod("background", &RMGGeneratorDecay0::SetBackground)
+      .SetGuidance("Set the isotope for the background mode of the BxDecay0 generator. E.g. 'Co60'")
+      .SetParameterName("isotope", false) // Do we want to add an enum for every supported isotope?
+      .SetToBeBroadcasted(true)
+      .SetStates(G4State_PreInit, G4State_Idle);
+
+  // Make one macro command for the dbd mode.
+  fUIMessenger = std::make_unique<BxMessenger>(this);
+}
+
+RMGGeneratorDecay0::BxMessenger::BxMessenger(RMGGeneratorDecay0* gen) : fGen(gen) {
+
+
+  fGeneratorCmd = new G4UIcommand("/RMG/Generator/BxDecay0/dbd", this);
+  fGeneratorCmd
+      ->SetGuidance("Set the isotope, process and energy level for the double beta decay mode of the BxDecay0 generator");
+
+  auto isotope = new G4UIparameter("isotope", 's', false);
+  isotope->SetGuidance(
+      "Set the isotope for the double beta decay"
+  ); // Do we want to add an enum for every supported isotope?
```

Comment:
```suggestion
  isotope->SetGuidance("Set the isotope for the double beta decay");
  std::string isot_cand;
  for (const auto& isot : bxdecay0::dbd_isotopes()) { isot_cand += std::string(isot) + ' '; }
  isotope->SetParameterCandidates(isot_cand.c_str());
```

bxdecay0 will complain (i.e., throw an exception) when it encounters an undefined isotope. However, that leads to a hang in MT mode as Geant4 doesn't like exceptions to be raised in MT mode before the run (yeah, I know...)

the candidates list should guard against that...